core/src/main/scala/org/json4s/Formats.scala

```diff
@@ -241,6 +259,11 @@ trait Formats extends Serializable { self: Formats =>
     }
 }
 
+trait RichSerializer[A] {
```

Comment:
`RichSerializer` is okay but I'm still thinking of another name for this... but I was not able to come up with any better ones. Any thoughts? > anyone

Reply:
Yeah I'm not in love with that name. Just not sure how to convey that it's the same thing as `Serializer` but with more type info.

Reply:
Tangentially related: An alternative to having `RichSerializer` might be to have a `BaseSerializer` which `Serializer` inherits from:
```scala
trait BaseSerializer[A] {
  def baseDeserialize(implicit format: Formats): PartialFunction[(ScalaType, JValue), A]
  def serialize(implicit format: Formats): PartialFunction[Any, JValue]
}

trait Serializer[A] extends BaseSerializer[A] {
  def deserialize(implicit format: Formats): PartialFunction[(TypeInfo, JValue), A]
  def baseDeserialize(implicit format: Formats): PartialFunction[(ScalaType, JValue), A] = {
    case (scalaType, obj) if deserialize.isDefinedAt(scalaType.typeInfo, obj) =>
      deserialize.apply(scalaType.typeInfo, obj)
  }
  def serialize(implicit format: Formats): PartialFunction[Any, JValue]
}
```
If that looks preferable, I could update this PR with the above snippet and update `wCustomSerializers: List[Serializer[_]] = self.customSerializers` to be `wCustomSerializers: List[BaseSerializer[_]] = self.customSerializers` (and make a few other similar changes) rather than adding `wRichSerializers`.

Reply:
I prefer just having `RichSerializer` over having the inheritance.